Makefile

```diff
@@ -35,4 +34,7 @@ re: fclean all
 test: $(NAME)
 	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJ) -o $(NAME) && ./$(NAME)
 
+valgrind: $(NAME)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJ) -o $(NAME) && valgrind ./$(NAME)

```

Comment:
Sugestão pro valgrind, usar a flag "-q" fazendo com que o valgrind fique silencioso e só mostre algo na tela se for algum erro (além do --leak-check=full)